### PR TITLE
[SDK-750] Update sample readme

### PR DESCRIPTION
**Pull request by skyflow-puneet**

### CODE REVIEW

**Review by Shaik-Luqmaan — APPROVED** [review]

File: samples/README.md [hunk 1]
```diff
@@ -1,13 +1,93 @@
...
+### Create the vault
+1. In a browser, navigate to Skyflow Studio and log in.
+2. Create a vault by clicking **Create Vault** > **Start With a Template** > **Quickstart vault**.
+3. Once the vault is created, click the gear icon and select **Edit Vault** Details.
+4. Note your Vault URL and Vault ID values, then click Cancel. You'll need these later.
+
+### Create a service account
+1. In the side navigation click, **IAM** > **Service Accounts** > **New Service Account**.
+2. For Name, enter **Test-Python-Sdk-Sample**. For Roles, choose the required roles for specific action.
+3. Click **Create**. Your browser downloads a **credentials.json** file. Keep this file secure, as you'll need it in the next steps.
```
Comment (manny-silva) [inline samples/README.md:23]: "3. Click **Create**. Your browser downloads a **credentials.json** file. Keep this file secure. You'll need it for each of the samples."

File: samples/README.md [hunk 2]
```diff
@@ -1,13 +1,93 @@
...
+
+### Different types of functionalities of Java-Sdk
```
Comment (manny-silva) [inline samples/README.md:25]: "## The samples"

File: samples/README.md [hunk 3]
```diff
@@ -1,13 +1,93 @@
...
+- [**Detokenize**](detokenize_sample.py)
```
Comment (manny-silva) [inline samples/README.md:26]: "### [**Detokenize**](detokenize_sample.py)"

File: samples/README.md [hunk 4]
```diff
@@ -1,13 +1,93 @@
...
+    - Detokenize the data token from the vault. 
+    - Make sure the token is of the data which exists in the Vault. If not so please make use of [insert_sample.py](insert_sample.py) to insert the data in the data and use this token for detokenization.
```
Comment (manny-silva) [inline samples/README.md:28]: "Detokenize a data token from the vault. Make sure the specified token is for data that exists in the vault. If you need a valid token, use [insert_sample.py](insert_sample.py) to insert the data, then use this data's token for detokenization."

File: samples/README.md [hunk 5]
```diff
@@ -1,13 +1,93 @@
...
+    - Configure
```
Comment (manny-silva) [inline samples/README.md:29]: "#### Configure"

File: samples/README.md [hunk 6]
```diff
@@ -1,13 +1,93 @@
...
+        - Replace **<YOUR_VAULT_ID>** with **VAULT ID**
+        - Replace **<YOUR_VAULT_URL>** with **VAULT URL**.
+        - Replace **<FIELD_NAME>** with **COLUMN NAME**.
+        - Replace **<TOKEN>** with **Data Token**.
+        - Replace **<YOUR_CREDENTIALS_FILE_PATH>** with relative  path of **SERVICE ACCOUNT CREDENTIAL FILE**.
```
Comment (manny-silva) [inline samples/README.md:34]: Change this whole list to an ordered list.

File: samples/README.md [hunk 7]
```diff
@@ -1,13 +1,93 @@
...
+    - Execution
```
Comment (manny-silva) [inline samples/README.md:35]: "#### Run the sample"

File: samples/README.md [hunk 8]
```diff
@@ -1,13 +1,93 @@
...
+        
+            python3 detokenize_smaple.py
```
Comment (manny-silva) [inline samples/README.md:37]:
```bash
python3 detokenize_smaple.py
```

File: samples/README.md [hunk 9]
```diff
@@ -1,13 +1,93 @@
...
+- [**GetById**](get_by_ids_sample.py)
```
Comment (manny-silva) [inline samples/README.md:38]: Same formatting comments as previous sample.

File: samples/README.md [hunk 10]
```diff
@@ -1,13 +1,93 @@
...
+        - Replace **<YOUR_VAULT_URL>** with **VAULT URL**.
+        - Replace **<SKYFLOW_ID1>** with **Skyflow Id 1**.
+        - Replace **<SKYFLOW_ID2>** with **Skyflow Id 2**.
+        - Replace **<SKYFLOW_ID3>** with **Skyflow Id 3**.
+        - Replace **<YOUR_CREDENTIALS_FILE_PATH>** with relative  path of **SERVICE ACCOUNT CREDENTIAL FILE**.
+        - Replace **<TABLE_NAME>** with **credit_cards**.
+    - Execution
+            
+            python3 get_by_ids_sample.py
+- [**Insert**](insert_sample.py)
```
Comment (manny-silva) [inline samples/README.md:51]: Same formatting comments as previous sample

File: samples/README.md [hunk 11]
```diff
@@ -1,13 +1,93 @@
...
+        - Replace **<YOUR_VAULT_ID>** with **VAULT ID**.
+        - Replace **<YOUR_VAULT_URL>** with **VAULT URL**.
+        - Replace **<YOUR_CREDENTIALS_FILE_PATH>** with relative  path of **SERVICE ACCOUNT CREDENTIAL FILE**.
+        - Replace **<TABLE_NAME>** with **credit_cards**.
+        - Replace **<FIELD_NAME>** with **column name**.
+        - Replace **<VALUE>** with **valid value corresponding to column name**.
+        - Execution
+            
+                python3 insert_sample.py
+- [**InvokeConnection**](invoke_connection_sample.py)
```
Comment (manny-silva) [inline samples/README.md:63]: Same formatting comments as previous sample

File: samples/README.md [hunk 12]
```diff
@@ -1,13 +1,93 @@
...
+        - Replace **<YOUR_CREDENTIALS_FILE_PATH>** with relative  path of **SERVICE ACCOUNT CREDENTIAL FILE**.
+        - Replace **<YOUR_CONNECTION_URL>** with **Connection url**.
+        - Replace **<YOUR_CONNECTION_BASIC_AUTH>** with **access token**.
+        - Replace value of **requestBody** with your's request body content.
+        - Replace value of **pathParams** with your's path params content.
+
+        - Execution
+            
+                python3 invoke_connection_sample.py
+- [**Service account token generation**](sa_token_Sample.py)
```
Comment (manny-silva) [inline samples/README.md:77]: Same formatting comments as previous sample

File: samples/README.md [hunk 13]
```diff
@@ -1,13 +1,93 @@
...
+    - generates SA Token using path of credentials file.
+    - Configure
+        - Replace **<YOUR_CREDENTIALS_FILE_PATH>** with relative  path of **SERVICE ACCOUNT CREDENTIAL FILE**.
+
+        - Execution
+            
+                python3 sa_token_sample.py
+
+- [**Generate Bearer Token From Credentails**](generate_bearer_token_from_creds_sample.py)
```
Comment (manny-silva) [inline samples/README.md:86]: Same formatting comments as previous sample

**Review by manny-silva — CHANGES_REQUESTED** [review]

File: samples/README.md [hunk 14]
```diff
@@ -1,13 +1,93 @@
-## Description
-These are the samples for skyflow-python. In order to use these samples, you must first:
+# PYTHON-SDK sample templates
+Use this folder to test the functionalities of PYTHON-SDK just by adding `VAULT-ID` `VAULT-URL` and `SERVICE-ACCOUNT` details at the required place.
 
-1. Install skyflow module using pip
-2. Replace placeholders like `<YOU_VAULT_ID>`, `<YOUR_VAULT_URL>` etc. in the sample file with your Vault ID, Vault URL etc.
+## Prerequisites
+- A Skylow account. If you don't have one, you can register for one on the [Try Skyflow](https://skyflow.com/try-skyflow) page.
+- Python 3.7.0 and above.
```
Comment (manny-silva) [inline samples/README.md:6]: - Python 3.7.0 or higher.

**Review by manny-silva — CHANGES_REQUESTED** [review]

**Comment by manny-silva** [comment]
Reformatted the sample readme with the following content. Would've committed directly myself, but I don't have edit access to this repo yet.

# Python SDK samples

Test the SDK by adding `VAULT-ID`, `VAULT-URL`, and `SERVICE-ACCOUNT` details in
the required places for each sample.

## Prerequisites

- A Skyflow account. If you don't have one, register for one on the
  [Try Skyflow](https://skyflow.com/try-skyflow) page.
- Python 3.7.0 or higher.

## Prepare

### Install the Python SDK

```bash
pip install skyflow
```

### Create the vault

1. In a browser, sign in to Skyflow Studio.
2. Create a vault by clicking **Create Vault** > **Start With a Template** >
   **Quickstart vault**.
3. Once the vault is ready, click the gear icon and select **Edit Vault Details**.
4. Note your **Vault URL** and **Vault ID** values, then click **Cancel**.
   You'll need these later.

### Create a service account

1. In the side navigation click, **IAM** > **Service Accounts** > **New Service Account**.
2. For **Name**, enter "SDK Sample". For **Roles**, choose **Vault Editor**.
3. Click **Create**. Your browser downloads a **credentials.json** file. Keep
   this file secure, as You'll need it for each of the samples.

## The samples

### [Get data by ID](./getByIDSample.py)

Get data using Skyflow IDs for the desired records.

#### Configure

Replace the following values in the sample file:

| Value                          | Description                                             |
| ------------------------------ | ------------------------------------------------------- |
| `<YOUR_VAULT_ID>`              | ID of your vault.                                       |
| `<YOUR_VAULT_URL>`             | URL of your vault.                                      |
| `<SKYFLOW_ID1>`                | Skyflow ID of the first record.                         |
| `<SKYFLOW_ID2>`                | Skyflow ID of the second record.                        |
| `<SKYFLOW_ID3>`                | Skyflow ID of the third record.                         |
| `<YOUR_CREDENTIALS_FILE_PATH>` | relative path to your service account credentials file. |
| `<TABLE_NAME>`                 | Name of the table to get data from.                     |

#### Run the sample

```bash
python3 getByIDSample.py
```

### [Insert data](./InsertSample.py)

Insert data in the vault.

#### Configure

Replace the following values in the sample file:

| Value                          | Description                                             |
| ------------------------------ | ------------------------------------------------------- |
| `<YOUR_VAULT_ID>`              | ID of your vault.                                       |
| `<YOUR_VAULT_URL>`             | URL of your vault.                                      |
| `<YOUR_CREDENTIALS_FILE_PATH>` | relative path to your service account credentials file. |
| `<TABLE_NAME>`                 | Name of the table to insert data into.                  |
| `<FIELD_NAME>`                 | Name of the column to insert data into.                 |
| `<VALUE>`                      | Valid value to insert into the corresponding column.    |

#### Run the sample

```bash
python3 InsertSample.py
```

### [Detokenize data](./detokenizeSample.py)

Detokenize a data token from the vault. Make sure the specified token is for
data that exists in the vault. If you need a valid token, use
[insert_sample.py](insert_sample.py) to insert the data, then use this data's
token for detokenization.

#### Configure

Replace the following values in the sample file:

| Value                          | Description                                             |
| ------------------------------ | ------------------------------------------------------- |
| `<YOUR_VAULT_ID>`              | ID of your vault.                                       |
| `<YOUR_VAULT_URL>`             | URL of your vault.                                      |
| `<YOUR_CREDENTIALS_FILE_PATH>` | relative path to your service account credentials file. |
| `<FIELD_NAME>`                 | Name of the column to insert data into.                 |
| `<TOKEN>`                      | Token for the data you want to detokenize.              |

#### Run the sample

```bash
python3 detokenizeSample.py
```

### [Invoke a connection](./invokeConnectionSample.py)

Skyflow Connections is a gateway service that uses Skyflow's underlying
tokenization capabilities to securely connect to first-party and third-party
services. This way, your infrastructure is never directly exposed to sensitive
data, and you offload security and compliance requirements to Skyflow.

#### Configure

Replace the following values in the sample file:

| Value                          | Description                                             |
| ------------------------------ | ------------------------------------------------------- |
| `<YOUR_VAULT_ID>`              | ID of your vault.                                       |
| `<YOUR_VAULT_URL>`             | URL of your vault.                                      |
| `<YOUR_CREDENTIALS_FILE_PATH>` | relative path to your service account credentials file. |
| `<YOUR_CONNECTION_URL>`        | URL of your connection.                                 |
| `<YOUR_CONNECTION_BASIC_AUTH>` | Access token for your connection.                       |
| `requestBody`                  | Your request body content.                              |
| `pathParams`                   | Your path parameters.                                   |

#### Run the sample

```bash
python3 invokeConnectionSample.py
```

### [Service account token generation](./SATokenSample.py)

Generates SA Token using path of credentials file.

#### Configure

Replace `<YOUR_CREDENTIALS_FILE_PATH>` with the relative path to your service account credentials file.

#### Run the sample

```bash
python3 SATokenSample.py
```

### [Generate Bearer Token](./generateBearerTokenFromCredsSample.py)

Generates SA Token using json content of credentials file.

#### Configure

Replace `credentials` with the content of service account credentials file.

#### Run the sample

```bash
python3 generateBearerTokenFromCredsSample.py
```

**Review by manny-silva — APPROVED** [review]

**Pull request merged**